Services/Catalogue/Mappers/ConfigurableProductsData.php

```diff
@@ -277,33 +281,9 @@ private function getCategoryArray($category)
             $parentUrlKey = '';
         }
 
-        if ($this->allStoreBaseUrls === null) {
-            $this->allStoreBaseUrls = $this->storeManager->getAllStoreBaseUrls();
-        }
-
-        if (!isset($this->currentStoreBaseUrl[$this->storeId])) {
-            $this->currentStoreBaseUrl[$this->storeId] = $this->storeManager->getCurrentStoreBaseUrl();
-            ;
-        }
-
-        $currentStoreBaseUrl = null;
-
-        if ($this->currentStoreBaseUrl[$this->storeId] &&
-            isset($this->currentStoreBaseUrl[$this->storeId]['base_url'])
-        ) {
-            $currentStoreBaseUrl = $this->currentStoreBaseUrl[$this->storeId];
-            $currentStoreBaseUrl = $currentStoreBaseUrl['base_url'];
-        }
-
-        $categoryUrl = $category->getUrl();
-        foreach ($this->allStoreBaseUrls as $store) {
-            if (strpos($categoryUrl, $store['base_url']) === 0 &&
-            isset($currentStoreBaseUrl) && $currentStoreBaseUrl !== null) {
-                $categoryUrl = str_replace($store['base_url'], '', $categoryUrl);
-                $categoryUrl = $currentStoreBaseUrl."".$categoryUrl;
-                break;
-            }
-        }
+        $storeBasedcategory = $this->categoryRepository->get($category->getId(), $this->storeId);
```

Comment:
No need of this,

Instead just try it on $category to set the scope and get the URL. If that doesn't work we need modify.
Services/Catalogue/CategoriesManager.php  -> fetchByIds